src/commands/defi/offchain_tip_bot/balance.ts

```diff
@@ -0,0 +1,109 @@
+import { Command } from "types/common"
+import { EmbedFieldData, Message } from "discord.js"
+import { BALANCE_GITBOOK, PREFIX, DEFI_DEFAULT_FOOTER } from "utils/constants"
+import {
+  emojis,
+  getEmoji,
+  getEmojiURL,
+  roundFloatNumber,
+  thumbnails,
+} from "utils/common"
+import Defi from "adapters/defi"
+import {
+  composeEmbedMessage,
+  getErrorEmbed,
+  justifyEmbedFields,
+} from "utils/discordEmbed"
+import { UserBalances } from "types/defi"
+
+const command: Command = {
+  id: "balancesoff",
+  command: "balancesoff",
+  brief: "Wallet balances",
+  category: "Defi",
+  run: async function balances(msg: Message) {
+    // case API return 500 or unpected result
+    const userId = msg.author.id
+    const res = await Defi.OffchainGetUserBalances({ userId })
+    if (!res.ok) {
+      return {
+        messageOptions: {
+          embeds: [getErrorEmbed({})],
+        },
+      }
+    }
+
+    // case data = null || []
+    if (!res.data || res.data.length === 0) {
+      const embed = composeEmbedMessage(msg, {
+        title: "Info",
+        description: `<@${msg.author.id}>, you have no balances.`,
```

Comment:
```suggestion
        description: `${msg.author}, you have no balances.`,
```

hàm `toString()` của class User của discord sẽ auto convert object thành format `<@id>`